### PR TITLE
Use token list in compile_eql rather than a string

### PR DESCRIPTION
**Pull request by tailhook**
This is a prerequisite for constant extraction PR. Not sure if this needs to be merged separately

### CODE REVIEW

File: edb/server/compiler/compiler.py [hunk 1]
```diff
@@ -166,7 +168,8 @@ def compile_edgeql_script(
     compiler._std_schema = std_schema
     compiler._bootstrap_mode = True
 
-    units = compiler._compile(ctx=ctx, eql=eql.encode())
+    tokens = _edgeql_rust.tokenize(eql)
+    units = compiler._compile(ctx=ctx, eql_tokens=tokens)
```
Comment (elprans) [inline edb/server/compiler/compiler.py:172]: nit: just call them `tokens`, it's clear that they're EdgeQL from the context.

File: edb/edgeql/parser/__init__.py [hunk 2]
```diff
@@ -19,9 +19,14 @@
 
 from __future__ import annotations
 
+from typing import TYPE_CHECKING, List
```
Comment (elprans) [inline edb/edgeql/parser/__init__.py:22]: Why not just `from typing import *` as everywhere else?

File: edb/server/compiler/compiler.py (hunk 1 above)
Comment (tailhook) [reply edb/server/compiler/compiler.py]: Fixed

File: edb/edgeql/parser/__init__.py (hunk 2 above)
Comment (tailhook) [reply edb/edgeql/parser/__init__.py]: Not sure. Probably just general allergy to `*`. Fixed.

**Pull request merged**